snapcraft/internal/repo/_deb.py

```diff
@@ -154,6 +155,51 @@ def _collected_sources_list(self):
         return _get_local_sources_list()
 
 
+class PackageNotFoundError(errors.PackageNotFoundError):
+
+    @property
+    def message(self):
+        message = super().message
+        # If the package was multiarch, try to help.
```

Comment:
This looks like it's building on what _used_ to be in `errors.PackageNotFoundError`. What is the rational for essentially redefining it here as opposed to moving this logic back into it?

Reply:
What do you mean by redefining?

My idea was to move the logic into the error because it needs to know
 * if the arch is valid
 * if the arch was registered
 * if there's sources for the arch
`_mark_install` doesn't need to make a distinction if we always want it to fail - previously I had the code in there because it would have had to continue iterating through the packages after automatically setting up the host system.

Also, we might be able to re-use the error with all its details. There's a case of `except PackageNotFoundError as e: raise BuildPackageNotFoundError` and another instance of raising PackageNotFoundError. Maybe this could be a follow-up.

Maybe having helper methods like `has_arch` still makes sense when I'm adding the container side of things, but aside from that I feel it would become less readable to do that in the same method as figuring out an error message.

Reply:
Ha, after typing all this, and reading your question once more, I realize you're probably referring to the fact that `_deb.PackageNotFoundError` is a subclass.
I did that because the logic is completely specific to Debian-based distros and the `is this a deb distro`-check is implied by being inside `_deb` as well. Presumably once there's a Fedora/dnf backend it will have completely different code (or none at all) depending on how architectures are handled there.

Reply:
Ah, good call, that makes sense indeed.